apps/sms/services/js/conversation/conversation_client.js

```diff
@@ -0,0 +1,72 @@
+/* global bridge,
+          streamClient,
+          Thread
+*/
+
+(function(exports) {
+  'use strict';
+
+  var debug = 0 ?
+    (arg1, ...args) => console.log('[ConversationClient] ' + arg1, ...args):
+    () => {};
+
+  /**
+   * Name of the service that is responsible for managing conversation.
+   * @type {string}
+   */
+  const SERVICE_NAME = 'conversation-service';
+
+  /**
+   * Reference to active bridge client instance.
+   *
+   * @type {Client}
+   */
+  var client;
+
+  /**
+   * ConversationClient is a wrapper around bridge client connected to the
+   * bridge service hosted in a Shared Worker.
+   * @type {Object}
+   */
+  var ConversationClient = {
+    /**
+     * Initialized conversation service client bridge.
+     */
+    init() {
+      var serviceEndpoint = new SharedWorker(
+        '/services/js/conversation/conversation_service.js'
+      );
+
+      serviceEndpoint.port.start();
+
+      client = bridge.client(SERVICE_NAME, serviceEndpoint.port).plugin(
+        streamClient
+      );
+    },
+
+    /**
+     * Retrieves all conversations and drafts.
+     * @param {Function} onConversationRetrieved Callback that is called for
+     * every retrieved conversation.
+     * @returns {Promise} Promise is resolved once all conversations are
+     * retrieved and rejected if conversation stream is closed unexpectedly.
+     */
+    getAllConversations(onConversationRetrieved) {
+      var getAllConversationsStream = client.stream('getAllConversations');
+
+      getAllConversationsStream.listen((thread) => {
+        debug('Conversation with id %s is retrieved from stream', thread.id);
+
+        onConversationRetrieved(new Thread(thread));
```

Comment:
> , but not sure for BroadcastChannels

I believe they're asyn as well. Just tried the following test case:
```
var sender = new BroadcastChannel('channel');

var receiver = new BroadcastChannel('channel');
receiver.onmessage = () => { 
	console.log('Time when on message handler invoked %s', new Date());
	var payload = Array.from({length: 9999999});
	payload.forEach(() => {});
	payload.forEach(() => {});
	console.log('Time when on message handler completed %s', new Date());
}

console.log('Time before message is posted %s', new Date());
sender.postMessage('a');
console.log('Time after message is posted %s', new Date());
```

Output is:
```
Time before message is posted Fri Jul 31 2015 19:53:52 GMT+0200 (CEST)
Time after message is posted Fri Jul 31 2015 19:53:52 GMT+0200 (CEST)
Time when on message handler invoked Fri Jul 31 2015 19:53:53 GMT+0200 (CEST)
Time when on message handler completed Fri Jul 31 2015 19:54:04 GMT+0200 (CEST)
```